backend/quiz.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+const (
+	QUIZ_SAMPLE_NAME = "/" + CATEGORY_SAMPLE_TEMPLATES + "/quiz/"
+)
+
+var rightAnswers []int
+
+func InitQuiz() {
+	rightAnswers = make([]int, 0)
+	rightAnswers = append(rightAnswers, 4, 2, 16, 1)
+	http.HandleFunc(QUIZ_SAMPLE_NAME+"submit", func(w http.ResponseWriter, r *http.Request) {
+		handlePost(w, r, submitQuiz)
+	})
+
+}
+
+func submitQuiz(w http.ResponseWriter, r *http.Request) {
+	EnableCors(w, r)
+	SetContentTypeJson(w)
+	response := ""
```

Comment:
Yes it is necessary, otherwise I get

> error.js:127 Failed to parse response JSON:: Unexpected end of JSON input
xd @ error.js:127
(anonymous) @ log.js:455
log.js:456 Uncaught SyntaxError: Failed to parse response JSON:: Unexpected end of JSON input _reported_